CMakeLists.txt

```diff
@@ -1,6 +1,20 @@
 cmake_minimum_required(VERSION 3.2)
 
-project(CBMC)
+
+# Grab the current CBMC version from config.inc
+# We do this so we have a matching cbmc version between the Makefile build and
+# the CMake build. This version info is useful for things like generating
+# packages (which usually need to contain version info in their metadata and
+# filenames)
+file(
+  STRINGS src/config.inc CBMC_VERSION
+  REGEX "CBMC_VERSION = (.*)")
+string(REGEX REPLACE "CBMC_VERSION = (.*)" "\\1" CBMC_VERSION ${CBMC_VERSION})
+
+project(CBMC VERSION ${CBMC_VERSION})
+
+# when config.inc changes we’ll need to reconfigure to check if the version changed
+set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/src/config.inc")
```

Comment:
Just a general query about this - when does config.inc get generated? IIRC it gets auto generated using the git info? If so, when does that change, and what is the effect here? I'm slightly worried about whether this means every time you make a local change and re-build, it _also_ causes a fresh re-configure? And if so, doesn't that also mean a fresh re-download of minisat?

Reply:
@chrisr-diffblue This string is static and currently only user set (plan is to autoincrement on release going forward). The git-tag string is _also_ used but only for use in `util/version.h`, which uses _both_ this info _and_ the git-tag.

The reason we keep the two separate is to support builds from source packages and shallow clones.